src/workerd/io/worker-interface.c++

```diff
@@ -34,12 +34,13 @@ public:
   }
 
   kj::Promise<void> connect(kj::StringPtr host, const kj::HttpHeaders& headers,
-      kj::AsyncIoStream& connection, ConnectResponse& response) override {
+      kj::AsyncIoStream& connection, ConnectResponse& response,
+      kj::HttpClient::ConnectSettings settings) override {
     KJ_IF_MAYBE(w, worker) {
-      return w->get()->connect(host, headers, connection, response);
+      return w->get()->connect(host, headers, connection, response, settings);
```

Comment:
Nit: `kj::mv(settings)`? I know it's not strictly needed now but if we add settings that need to be moved later it helps.

Reply:
Might even be worth declaring the settings object move-only now, to avoid breakage in the future.

Reply:
making it move-only seems painful, if we add settings that need to be moved we can make it move-only then.